frontend/my-app/src/app/shopping-item/shopping-item.component.ts

```diff
@@ -0,0 +1,24 @@
+import {Component, Input, OnInit, Output, EventEmitter} from '@angular/core';
+import {ShoppingItem} from './shopping-item';
+
+
+@Component({
+  selector: 'app-shopping-item',
+  templateUrl: './shopping-item.component.html',
+  styleUrls: ['./shopping-item.component.css']
+})
+export class ShoppingItemComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
+  @Input() item: ShoppingItem;
+  @Output() onRemove = new EventEmitter<never>();
```

Comment:
new EventEmitter<ShoppingItem>();
i wtedy można zrobić tak:
this.onRemove.emit(this.item);